provider/gce/google/errors.go

```diff
@@ -120,3 +120,11 @@ var AuthorisationFailureStatusCodes = map[int][]string{
 	// https://tools.ietf.org/html/rfc6749#section-5.2
 	http.StatusBadRequest: {"Bad Request"},
 }
+
+// IsNotFound reports whether err contains `not found'.
+func IsNotFound(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(err.Error(), " not found")
```

Comment:
This is pretty rough - is there anything more specific than this you can use to detect errors from `RemoveInstance`, `RemoveFirewall` and `GetFirewalls`?

Reply:
As per my reply to Harry's comment, it actually covers all cases - both scenarios that come from google and the errors.IsNotFound...

I am running  last test with a more precise implementation:
```
	if gerr, ok := errors.Cause(err).(*googleapi.Error); ok {
		return gerr.Code == http.StatusNotFound || strings.Contains(gerr.Error(), "not found")
	}
	return errors.IsNotFound(err)
```
but so far, my original one is the cleanest and quickest.

Reply:
So after further tests, the answer is a no. This covers all the cases.